app/controllers/webhooks/stripe/base_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Webhooks::Stripe::BaseController < ApplicationController
+  include StripeWebhookEvents
+
+  skip_before_action :verify_authenticity_token
+  protect_from_forgery with: :null_session
+
+  before_action :authenticate
+  before_action :validate_event_type
+  before_action :validate_event_environment if Rails.env.production?
+
+  rescue_from Stripe::APIConnectionError, with: :stripe_error
+  rescue_from Stripe::StripeError, with: :stripe_error
+
+  def stripe_error
+    head status: 200
```

Comment:
I don't think we should be returning 200s for _every_ Stripe error. Think of the scenarios:

1. We receive a webhook from Stripe but then try to request the `Stripe::Event` to verify the authenticity of the event. That request times out or raises an error. The event doesn't process. We should return a 4XX or 5XX error to let Stripe know we weren't able to process the event.

2. Possibly a made up scenario but consider this: we receive a bad request where an event hits our webhook endpoint with a `Stripe::Event` that doesn't exist on Stripe anymore. Stripe returns a 404 when trying to request the event data from their API. We should return a 2XX to indicate that we don't want to process the request anymore.

3. Our db is borked and when trying to save a record to the db, we raise an error. A 5XX error should be returned back to Stripe.

So, in some cases we don't want to return a 2XX response because the event was still not processed yet. Returning a 4XX or 5XX response will ensure that the event will eventually get processed by Stripe retrying.

Another thing to consider is offloading this work off to a Sidekiq job that we ourselves can retry as many times as we want. This gives us the flexibility to retry as many times as we see fit. In that scenario, we'd always respond back with a 2XX. Just something to consider.